transform/binary-plugin/binary_plugin.go

```diff
@@ -0,0 +1,78 @@
+package binary_plugin
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+
+	"github.com/konveyor/crane-lib/transform"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type BinaryPlugin struct {
+	commandRunner
+	log logrus.FieldLogger
+}
+
+func NewBinaryPlugin(path string) *BinaryPlugin {
+	return &BinaryPlugin{commandRunner: &binaryRunner{path: path}, log: logrus.New().WithField("path", path)}
+}
+
+func (b *BinaryPlugin) Run(u *unstructured.Unstructured) (transform.PluginResponse, error) {
+	p := transform.PluginResponse{}
+
+	out, errBytes, err := b.commandRunner.Run(u, b.log)
+	if err != nil {
+		b.log.Errorf("error running the plugin command")
+		return p, fmt.Errorf("error running the plugin command: %v", err)
+	}
+
+	if len(errBytes) != 0 {
+		b.log.Errorf("error from plugin binary")
+		return p, fmt.Errorf("error from plugin binary: %s", string(errBytes))
+	}
+
+	err = json.Unmarshal(out, &p)
+	if err != nil {
+		b.log.Errorf("unable to decode json sent by the plugin")
+		return p, fmt.Errorf("unable to decode json sent by the plugin: %s, err: %v", string(out), err)
+	}
+
+	return p, nil
+}
+
+type commandRunner interface {
```

Comment:
do we have to pass a logger? If we do why should that not be on the main interface?

Reply:
I have created a separate interface because this makes testing easier https://github.com/konveyor/crane-lib/pull/6/files#diff-6150fb93d22d7a0c24d49a757cdd9e8c86f8eac4dbe54e4532ceadbf2d9090c7R13

Thought of passing a logger instead of making it a field in the struct so that that binaryRunner struct is thread-safe for future version when this can be made parallel with goroutines.

The main interface can still hide the logger in implementation structs instead of passing the value to keep the interface surface small. Either way works for me honestly